pkg/plugins/client.go

```diff
@@ -138,7 +147,23 @@ func (c *Client) callWithRetry(serviceMethod string, data io.Reader, retry bool)
 			return nil, err
 		}
 
+		var callbacks []func()
+		if len(reqOpts) > 0 {
+			callbacks = make([]func(), 0, len(reqOpts))
+		}
+
+		for _, o := range reqOpts {
+			var cb func()
+			req, cb = o(req)
+			if cb != nil {
+				callbacks = append(callbacks, cb)
+			}
+		}
+
 		resp, err := c.http.Do(req)
+		for _, cb := range callbacks {
```

Comment:
Is it documented that request context does not have any effect after `Do` returns? Technically the request is still active at this point.